pravega-client-examples/src/test/java/io/pravega/utils/SetupUtils.java

```diff
@@ -0,0 +1,353 @@
+     *
+     * @throws Exception on any errors.
+     */
+    public void createTestStream(final String streamName, final int numSegments)
+            throws Exception {
+        Preconditions.checkState(this.started.get(), "Services not yet started");
+        Preconditions.checkNotNull(streamName);
+        Preconditions.checkArgument(numSegments > 0);
+
+        @Cleanup
+        StreamManager streamManager = StreamManager.create(getClientConfig());
+        streamManager.createScope(this.scope);
+        streamManager.createStream(this.scope, streamName,
+                StreamConfiguration.builder()
+                        .scalingPolicy(ScalingPolicy.fixed(numSegments))
+                        .build());
+        log.info("Created stream: " + streamName);
+    }
+
+    /**
+     * Get the stream.
+     *
+     * @param streamName     Name of the test stream.
+     *
+     * @return a Stream
+     */
+    public Stream getStream(final String streamName) {
+        return Stream.of(this.scope, streamName);
+    }
+
+    /**
+     * A gateway interface to Pravega for integration test purposes.
+     */
+    private interface PravegaGateway {
+        /**
+         * Starts the gateway.
+         */
+        void start() throws Exception;
+
+        /**
+         * Stops the gateway.
+         */
+        void stop() throws Exception;
+
+        /**
+         * Gets the client configuration with which to connect to the controller.
+         */
+        ClientConfig getClientConfig();
+    }
+
+    class InProcPravegaGateway implements PravegaGateway {
+
+        // The pravega cluster.
+        private InProcPravegaCluster inProcPravegaCluster = null;
+
+        @Override
+        public void start() throws Exception {
+            log.info("Starting gateway");
+            int zkPort = PortUtils.getAvailableListenPort();
+            int controllerPort = PortUtils.getAvailableListenPort();
+            int hostPort = PortUtils.getAvailableListenPort();
+            int restPort = PortUtils.getAvailableListenPort();
+
+            log.info("Building");
+            this.inProcPravegaCluster = InProcPravegaCluster.builder()
+                    .isInProcZK(true)
+                    .secureZK(enableTls) //configure ZK for security
+                    .zkUrl("localhost:" + zkPort)
+                    .zkPort(zkPort)
+                    .isInMemStorage(true)
+                    .isInProcController(true)
+                    .controllerCount(1)
+                    .restServerPort(restPort)
+                    .enableRestServer(enableRestServer)
+                    .isInProcSegmentStore(true)
+                    .segmentStoreCount(1)
+                    .containerCount(4)
+                    .enableMetrics(false)
+                    .enableAuth(enableAuth)
+                    .enableTls(enableTls)
+                    //.certFile(getFileFromResource(CERT_FILE))   // pravega #2519
```

Comment:
Do we need to clean this up?

Reply:
Fixed. I removed all commented and unused code.